app/jobs/course/course_github_repo_get_commits.rb

```diff
@@ -4,9 +4,134 @@ class CourseGithubRepoGetCommits < CourseGithubRepoJob
     @job_short_name = "course_github_repo_get_commits"
     @job_description = "Get commits for this repo"
     
+
     def attempt_job(options)
-      "Job Complete; Getting commits not yet implemented."
+      results = perform_graphql_query(@github_repo.name,@course.course_organization)
+      commits = results[:data][:repository][:ref][:target][:history][:edges]
+      result = store_commits_in_database(commits)
+      "Job Complete; Retrieved #{commits.length} commits for Course #{@course.name} for Repo #{@github_repo.name}. Stored #{result["total_new_commits"]} new commits, Updated #{result["total_updated_commits"]} existing commits in database."
+    end  
+
+    def store_commits_in_database(commits)
+      total_new_commits = 0
+      total_updated_commits = 0
+      commits.each{ |c|
+        existing_commit = RepoCommitEvent.where(commit_hash: c[:node][:oid]).first
+        if existing_commit
+          total_updated_commits += update_one_commit(existing_commit, c)
+        else
+          total_new_commits += store_one_commit_in_database(c) 
+        end
+      }
+      result = { 
+        "total_new_commits" => total_new_commits,
+        "total_updated_commits" => total_updated_commits
+      }
+      result
     end
+
+    def store_one_commit_in_database(c)
     
+      commit = RepoCommitEvent.new
+      commit.files_changed = c[:node][:changedFiles]
+      commit.message = c[:node][:message]
+      commit.commit_hash =  c[:node][:oid]
+      commit.url =  c[:node][:url]
+      commit.commit_timestamp =  c[:node][:author][:date]
+      commit.github_repo = @github_repo
+      commit.branch = "master"
+      begin  # "try" block
+        uid = c[:node][:author][:user][:databaseId]
+        commit.roster_student = lookup_roster_student_by_github_uid(uid)
+      rescue # optionally: `rescue Exception => ex`
+        uid = ""
+        commit.roster_student = nil
+      end 
```

Comment:
(This and below comment are not necessary changes, just a couple tips for writing idiomatic Ruby code.)

If the begin/rescue block here is to catch one level of the `c` hash being nil and blowing up, there's a handy Ruby method `#dig` for hashes that lets you get a deeply nested value from a hash without painful nil checking or exception control flow. More info here: https://prathamesh.tech/2020/03/18/diggin-through-hashes-and-arrays-in-ruby/